src/cli.rs

```diff
@@ -25,9 +25,9 @@ pub(crate) struct Cli {
     /// Seconds to wait for output before killing the task
     #[arg(long)]
     pub(crate) task_output_timeout: Option<u64>,
-    /// Slack Webhook for notification
-    #[arg(long, value_parser(Url::from_str), env = "HEALTH_CHECK_SLACK_WEBHOOK")]
-    pub(crate) slack_webhook: Option<Url>,
+    /// Notification hook
+    #[command(flatten)]
+    pub(crate) notification_hook: NotifyHook,
```

Comment:
I'm confused by this change. I would have expected we'd change the type instead to get rid of the `Option`. Why do we need this helper type?

Reply:
Doh, I read more carefully:

> Making it as non optional would indicate that this tool is tied to slack.

We're beyond my familiarity with clap in that case. But given that we are only using Slack right now, I'd follow YAGNI and just strip off `Option` until we add other environments.

Reply:
Sure, updated. Thanks!

Reply:
Same question as before though: why do we need the extra `NotifyHook` data type? Can we keep the original `slack_webhook` field and change its type to `Url` instead of `Option<Url>`?

Reply:
Ah, I see what you mean. Fixed, thanks!